pkg/admin/data/data_manager.go

```diff
@@ -0,0 +1,105 @@
+package data
+
+import (
+	"bytes"
+	"fmt"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/openshift/microshift/pkg/config"
+	"github.com/openshift/microshift/pkg/util"
+	"k8s.io/klog/v2"
+)
+
+var (
+	cpArgs = []string{
+		"--verbose",
+		"--recursive",
+		"--preserve",
+		"--reflink=auto",
+	}
+)
+
+func NewManager(storage StoragePath) (*manager, error) {
+	if storage == "" {
+		return nil, &EmptyArgErr{argName: "storage"}
+	}
+	return &manager{storage: storage}, nil
+}
+
+var _ Manager = (*manager)(nil)
+
+type manager struct {
+	storage StoragePath
+}
+
+func (dm *manager) GetBackupPath(name BackupName) string {
+	return filepath.Join(string(dm.storage), string(name))
+}
+
+func (dm *manager) BackupExists(name BackupName) (bool, error) {
+	return pathExists(dm.GetBackupPath(name))
+}
+
+func (dm *manager) Backup(name BackupName) error {
+	klog.InfoS("Backing up the data",
+		"storage", dm.storage, "name", name, "data", config.DataDir)
+
+	if name == "" {
+		return &EmptyArgErr{"name"}
+	}
+
+	if found, err := pathExists(string(dm.storage)); err != nil {
+		return err
+	} else if !found {
+		if makeDirErr := util.MakeDir(string(dm.storage)); makeDirErr != nil {
+			return fmt.Errorf("making %s directory failed: %w", dm.storage, makeDirErr)
+		}
+		klog.InfoS("Backup storage directory created", "path", dm.storage)
+	} else {
+		klog.InfoS("Backup storage directory already existed", "path", dm.storage)
```

Comment:
This indicates an attempt to write over an existing backup, right? I would expect that to be an error.

Reply:
Not really, that's the "parent directory" where backups are stored. Maybe you have how can we name it better ("backup storage") or maybe just simply delete this log (and keep only one about creating /var/lib/microshift-backups/)?

Reply:
Ah, I see what's going on now.